packages/db-mongodb/src/find.ts

```diff
@@ -55,7 +55,7 @@ export const find: Find = async function find(
     useEstimatedCount,
   }
 
-  if (!useEstimatedCount && this.disableIndexHints !== true) {
+  if (!useEstimatedCount && Object.keys(query).length === 0 && this.disableIndexHints !== true) {
     // Improve the performance of the countDocuments query which is used if useEstimatedCount is set to false by adding a hint.
```

Comment:
I'd change the comment to something like

// Improve the performance of the countDocuments query which is used if useEstimatedCount is set to false by adding a hint. By default, if no hint is provided, MongoDB does not use an indexed field to count the returned documents, which makes queries very slow. This only happens when no query (filter) is provided. If one is provided, it uses the correct indexed field

as, without knowing the context, it's unclear why this code is needed